File: PharmaCity.Angular/src/models/Petition.ts
import { State } from "./State";

export class Petition {

    id: number;
    medicineCode: string;
    quantity: number;
    state: State;
    stateString: string;

    constructor(){
        this.id = 0;
        this.medicineCode = "";
        this.quantity = 0;
        this.state = State.Inactive;
        this.stateString = "";
    }
}

File: PharmaCity.System/PharmaCity.Domain/Petition.cs
using System;

namespace PharmaCity.Domain
{
    public class Petition
    {
        private int _quantity;

        public int Id { get; set; }
        public string MedicineCode { get; set; }
        public int Quantity 
        { 
            get => _quantity;

            set
            {
                if (value > 0)
                {
                    _quantity = value;
                }
                else
                {
                    throw new InvalidOperationException("La cantidad de la peticion debe ser mayor a 0");
                }
                
            }
        }
        public Pharmacy Pharmacy { get; set; }
        public State State { get; set; }
    }
}